modules/siddhi-core/src/main/java/org/wso2/siddhi/core/SiddhiManager.java

```diff
@@ -191,6 +212,41 @@ public void persist() {
     }
 
     /**
+     * Method to get the current snapshot of all siddhi applications.
+     *
+     * @return map with siddhi app runtime name as key and snapshot in byte array as value.
+     */
+    public Map<String, byte[]> getSnapshot() {
+        Map<String, byte[]> snapshotMap = new HashMap<>();
+        for (SiddhiAppRuntime siddhiAppRuntime : siddhiAppRuntimeMap.values()) {
+            byte[] snapshot = siddhiAppRuntime.snapshot();
+            snapshotMap.put(siddhiAppRuntime.getName(), snapshot);
+        }
+        return snapshotMap;
+    }
+
+    /**
+     * Method to restore a given snapshot for all siddhi applications.
+     *
+     * @param snapshotMap map with siddhi app runtime name as key and snapshot in byte array as value.
+     */
+    public void restoreState(Map<String, byte[]> snapshotMap) {
+        for (SiddhiAppRuntime siddhiAppRuntime : siddhiAppRuntimeMap.values()) {
+            siddhiAppRuntime.restore(snapshotMap.get(siddhiAppRuntime.getName()));
+        }
+    }
+
+    /**
+     * Method to restore a given snapshot of a given siddhi application.
+     *
+     * @param siddhiAppRuntimeName the name of the siddhi app whose snapshot is to be restored.
+     * @param snapshot the snapshot of the siddhi app that should be restored.
+     */
+    public void restoreState(String siddhiAppRuntimeName, byte[] snapshot) {
+        siddhiAppRuntimeMap.get(siddhiAppRuntimeName).restore(snapshot);
```

Comment:
Won't there be a null pointer here if the siddhiAppRuntime is not available  ?

Reply:
Yes. Will do null check